docs/design/storagenode-downtime-tracking.md

```diff
@@ -0,0 +1,221 @@
+
+We need to enable back uptime disqualification, however before, we need to track in a fair way the amount of time Storage Nodes are offline.
+
+This document presents a solution for tracking  Storage Node offline time, but it's out of the scope of this document how Storage Node reputation is affected by it.
+
+The presented solution considers the requirements imposed by the [Kademlia removal design document](kademia-removal.md), where the Storage Nodes must contact the Satellite when they start and after every hour.
+
+## Design
+
+For tracking Storage Nodes offline time we need:
+
+- A new SQL database table to store offline time.
+- Detecting which ones are offline.
+- Roughly estimate how long they are offline.
+
+An _uptime check_ referenced in this section is the connection initiated by the Satellite to any Storage Node in the same way that's described in [Network refreshing section of the Kademlia removal design document](kademia-removal.md#network-refreshing).
+
+__NOTE__ the SQL code in this section is illustrative for explaining the algorithm concisely.
+
+### Database
+
+The following new SQL database table will store Storage Nodes offline time.
+
+```sql
+CREATE TABLE nodes_offline_time(
+    node_id BYTEA NOT NULL,
+    tracked_at timestamp with time zone NOT NULL,
+    seconds integer,                                                    -- Measured number of seconds being offline
+    CONSTRAINT nodes_offline_time__pk PRIMARY KEY(node_id, tracked_at)
+)
+```
+
+The current Satellite database has the table `nodes`. For the offline time calculation we use the following columns:
+
+- `last_contact_success`
+- `last_contact_failure`
+- `total_uptime_count`
+- `uptime_success_count`
+
+### Detecting offline nodes
+
+Per [Kademlia Removal design document](https://github.com/storj/storj/blob/master/docs/design/kademlia-removal.md#network-refreshing), any Storage Node has to ping the Satellite every hour and when it starts up. We have to contact the Storage Nodes which the Satellite hasn't registered a ping for more than one hour to determinie if they are offline or for whatever reason their ping didn't reach the Satellite.
+
+For checking those Storage Nodes, an independent process runs in a configurable interval of time the following query:
+
+```sql
+SELECT
+FROM nodes
+WHERE
+    last_contact_success < (now()  - 1h) AND
+    last_contact_success > last_contact_failure AND
+    disqualified IS NULL
+ORDER BY
+    last_contact_success ASC
+LIMIT 1
+```
+
+The process runs the query repeatable until it doesn't return records then it ends and will run again in the next interval. The query only gets one record at the time for not holding a record set during the process.
+
+For each node, the Satellite performs an _uptime check_.
+
+* On success
+    ```sql
+        UPDATE nodes
+        SET
+            last_contact_success = now(),
```

Comment:
Wondering whether this should be `last_contact_success = max(last_contact_success, now()),` which would play more nicely with batching.

Reply:
yes, that's much better.
And the same applies for the _on success_  of the _Estimating offline time_ section

Reply:
Applied.